rskj-core/src/main/java/org/ethereum/crypto/EthereumIESEngine.java

```diff
@@ -250,7 +243,9 @@ private byte[] encryptBlock(
             k2A = k2;
         }
         mac.init(new KeyParameter(k2A));
-        mac.update(iv, 0, iv.length);
+        if(iv != null) {
+            mac.update(iv, 0, iv.length);
+        }
```

Comment:
```suggestion
        mac.update(Objects.requireNonNull(iv), 0, iv.length);
```
before the change `NullPointerException` was being thrown if `null`. I guess we should preserve this.